### PR TITLE
Introduce role priorities

### PR DESCRIPTION
**Pull request by pkprzekwas**
**What does this PR do / Why do we need it**:
As discussed on sig api-ui, replaced prioritisation of user project binding with most permissive role.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/api/v1/types.go [hunk 1]
```diff
@@ -589,28 +597,23 @@ func ConvertInternalUserToExternal(internalUser *kubermaticv1.User, includeSetti
...
+			if pg.ID == binding.Spec.ProjectID {
+				// Check if role from group binding is more permissive.
+				if RolePriority[binding.Spec.Role] > RolePriority[pg.GroupPrefix] {
+					overwriteRole = true
+				}
 				break
 			}
 		}
-		if !bindingAlreadyExists {
+		if overwriteRole {
```
Comment (lsviben) [inline pkg/api/v1/types.go]: I think now you are adding projects only if the role priority is higher. If a userprojectbinding does not exist, you are not adding a project through GPBs

Comment (pkprzekwas) [reply pkg/api/v1/types.go]: Oh man, u r right. Let me fix this.

**Pull request merged**